flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdDistinctRowCount.scala

```diff
@@ -223,7 +223,7 @@ class FlinkRelMdDistinctRowCount private extends MetadataHandler[BuiltInMetadata
         val groupKeyOfCurrentProject = new JArrayList[Int]()
         groupKeySkipExpandId.foreach { key =>
           project.get(key) match {
-            case literal: RexLiteral if literal.isNull => // do nothing
+            case literal: RexLiteral if literal.isNull || literal.isAlwaysFalse => // do nothing
```

Comment:
there are some corner case ?

Reply:
It's unnecessary now, just a patch for the boolean type literal which introduced from the previous offline discussion.
I'll remove it.